distiller/policy.py

```diff
@@ -202,15 +203,19 @@ def on_minibatch_end(self, model, epoch, minibatch_id, minibatches_per_epoch, ze
 
 
 class LRPolicy(ScheduledTrainingPolicy):
-    """ Learning-rate decay scheduling policy.
+    """Learning-rate decay scheduling policy.
 
     """
     def __init__(self, lr_scheduler):
         super(LRPolicy, self).__init__()
         self.lr_scheduler = lr_scheduler
 
-    def on_epoch_begin(self, model, zeros_mask_dict, meta):
-        self.lr_scheduler.step()
+    def on_epoch_begin(self, model, zeros_mask_dict, meta, **kwargs):
+        if isinstance(self.lr_scheduler, torch.optim.lr_scheduler.ReduceLROnPlateau):
+            # Note: ReduceLROnPlateau doesn't inherit from _LRScheduler
+            self.lr_scheduler.step(**kwargs)
```

Comment:
Elsewhere we use `meta['current_epoch]` to get the epoch number, let's be consistent with that. The other 2 comments are related to this as well.
```suggestion
            self.lr_scheduler.step(kwargs['metrics'], meta['current_epoch'])
```